### PR TITLE
Node: p2p.Run interface change

### PR DESCRIPTION
**Pull request by bruce-riley**
This PR changes `p2p.Run` to take a parameter structure, rather than over two dozen individual parameters. The idea is that most of those parameters only have meaning to the guardian, so we should not require applications like the spy, fly and WormholeScan to setup numerous parameters that they don't need, just to suit the API. Additionally, this will allow us to add extra parameters that are of use only to the guardian without having to modify those other applications.

Once this PR is approved and merged, this new functionality with simplify two upcoming guardian features:

- Observation Batching
- Gossip Topic Split

Note that this PR follows a typical go approach for passing in optional parameters using `With` functions. For instance, this same approach is used in `libp2p`, which served as a pattern for this PR.

### CODE REVIEW

File: node/pkg/node/options.go [hunk 1]
```diff
@@ -55,33 +55,38 @@ func GuardianOptionP2P(p2pKey libp2p_crypto.PrivKey, networkId, bootstrapPeers,
...
+			params, err := p2p.NewRunParams(
+				bootstrapPeers,
+				networkId,
 				p2pKey,
-				g.gk,
 				g.gst,
-				networkId,
-				bootstrapPeers,
-				nodeName,
-				disableHeartbeatVerify,
```
Comment (evan-gray) [inline node/pkg/node/options.go:70]: I think this one got lost. Not sure if that was intentional.

**Review by evan-gray — APPROVED** [review]

File: node/pkg/node/options.go (hunk 1 above)
Comment (bruce-riley) [reply node/pkg/node/options.go:70]: Good catch. I added it to `WithGuardianOptions`.

**Review by evan-gray — APPROVED** [review]

**Review by ftocal — APPROVED** [review]

**Review by walker-16 — APPROVED** [review]

File: node/pkg/p2p/run_params.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+		p.disableHeartbeatVerify = disableHeartbeatVerify
+		return nil
+	}
+}
+
+// WithGuardianOptions is used to set options that are only meaningful to the guardian.
+func WithGuardianOptions(
+	nodeName string,
+	gk *ecdsa.PrivateKey,
+	obsvC chan<- *common.MsgWithTimeStamp[gossipv1.SignedObservation],
```
Comment (panoel) [inline node/pkg/p2p/run_params.go:150]: Isn't this one set with WithSignedObservationListener?

Comment (panoel) [reply node/pkg/p2p/run_params.go:150]: Is this so that you don't have to call all the functions and just pass all the values in?

Comment (bruce-riley) [reply node/pkg/p2p/run_params.go:150]: Yes, but that is meant to be used by clients. I wanted `WithGuardianOptions` to take everything that the guardian requires.

File: node/pkg/p2p/run_params.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+	ibcFeaturesFunc func() string,
+	gatewayRelayerEnabled bool,
+	ccqEnabled bool,
+	signedQueryReqC chan<- *gossipv1.SignedQueryRequest,
+	queryResponseReadC <-chan *query.QueryResponsePublication,
+	ccqBootstrapPeers string,
+	ccqPort uint,
+	ccqAllowedPeers string,
+) RunOpt {
+	return func(p *RunParams) error {
```
Comment (panoel) [inline node/pkg/p2p/run_params.go:168]: Do you have to return an error type for this?  It's always nil.

**Review by panoel — APPROVED** [review]

File: node/pkg/p2p/run_params.go (hunk 3 above)
Comment (bruce-riley) [reply node/pkg/p2p/run_params.go:168]: That is the signature of `RunOpt`. Although none of those return an error at this point, some might some day, so I would like to leave it. Also, that is the pattern that I copied from libp2p...

**Pull request merged**